src/components/structures/login/Registration.js

```diff
@@ -228,11 +232,15 @@ module.exports = React.createClass({
                 break; // NOP
             case "Register.START":
             case "Register.STEP_m.login.dummy":
+                // NB. Our 'username' prop is specifically for upgrading
+                // a guest account
                 registerStep = (
                     <RegistrationForm
                         showEmail={true}
-                        defaultUsername={this.props.username}
-                        defaultEmail={this.props.email}
+                        defaultUsername={this.state.formVals.username}
+                        defaultEmail={this.state.formVals.email}
+                        defaultPassword={this.state.formVals.password}
```

Comment:
I'm struggling to identify when these will actually be set *on startup*. When this component is loaded, `formVals === { email: <something> }`, with an empty `username` and `password`. Those fields are only set when the user hits the submit button, but by that time the whole point of the defaults (the thing which appears when the form is loaded) is moot. We don't seem to persist `formVals` either so I can't see how refreshing would Do The Right Thing.

What am I missing here?